### PR TITLE
Adds example for reduce operator #25468

### PR DESCRIPTION
**Pull request by mehdimousavi1995**
Refs #25468

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @mehdimousavi1995,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by raboof** [comment]
Hello @mehdimousavi1995, thanks for starting work on this! Unfortunately, your commit doesn't appear to contain any changes. Perhaps you have some changes that you have not 'added' to any commit yet?

**Comment by mehdimousavi1995** [comment]
Sorry, i've forgotten to add changes. I've fixed the problem

**Comment by raboof** [comment]
OK TO TEST

**Review by raboof — APPROVED** [review]
Nice one, thanks!

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/SinkSpec.scala [hunk 1]
```diff
@@ -324,6 +324,18 @@ class SinkSpec extends StreamSpec with DefaultTimeout with ScalaFutures {
     }
   }
 
+  "The reduce sink" must {
+    "sum up 1 to 10 correctly" in {
+      //#reduce-operator-example
+      val source = Source(1 to 10)
+      val rightValue = source.toMat(Sink.reduce((a: Int, b: Int) ⇒ a + b))(Keep.right).run
```
Comment (johanandren) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/SinkSpec.scala]: If you put an explicit type param on reduce instead you can do `Sink.reduce[Int]((a, b) => a + b), and then use `runWith(sink)` it returns the materialized value from the sink without having to use the `Keep.right` part (last part is true for Java as well).

**Review by johanandren — COMMENTED** [review]
A couple of tiny simplifications possible!

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/SinkSpec.scala (hunk 1 above)
Comment (mehdimousavi1995) [reply akka-stream-tests/src/test/scala/akka/stream/scaladsl/SinkSpec.scala]: done.

**Review by johanandren — APPROVED** [review]
LGTM, thanks!

**Pull request merged**